src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -37,6 +37,11 @@
     const ALLOW_EXTRA_ATTRIBUTES = 'allow_extra_attributes';
     const DISABLE_TYPE_ENFORCEMENT = 'disable_type_enforcement';
 
+    /**
+     * @var callable
+     */
+    protected $maxDepthHandler;
```

Comment:
should be private, isn't it?

Reply:
The circular reference handler is protected, so it can be set in a subclass. I've made this one protected too for the sake of consistency.

Reply:
If it can be private, it should be. Consistency shouldn't win over ease of maintenance.

Reply:
`$classDiscriminatorResolver` is also protected (it has been introduced only in 4.1), should we also change its visibility?

Reply:
yes please (same PR to reduce process overhead if possible :) )